src/Illuminate/Log/LoggerConfiguration.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Illuminate\Log;
+
+use InvalidArgumentException;
+use Monolog\Logger as Monolog;
+
+trait LoggerConfiguration
+{
+    /**
+     * The Log levels.
+     *
+     * @var array
+     */
+    protected $levels = [
+        'debug' => Monolog::DEBUG,
+        'info' => Monolog::INFO,
+        'notice' => Monolog::NOTICE,
+        'warning' => Monolog::WARNING,
+        'error' => Monolog::ERROR,
+        'critical' => Monolog::CRITICAL,
+        'alert' => Monolog::ALERT,
+        'emergency' => Monolog::EMERGENCY,
+    ];
+
+    /**
+     * Parse the string level into a Monolog constant.
+     *
+     * @param  array  $config
+     * @return int
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected function level(array $config)
+    {
+        $level = $config['level'] ?? 'debug';
+
+        if (isset($this->levels[$level])) {
+            return $this->levels[$level];
+        }
+
+        throw new InvalidArgumentException('Invalid log level.');
+    }
+
+    /**
+     * Extract the log channel from the given configuration.
+     *
+     * @param  array  $config
+     * @return string
+     */
+    protected function parseChannel(array $config)
+    {
+        if (! isset($config['name'])) {
+            return $this->app->bound('env') ? $this->app->environment() : 'production';
```

Comment:
using `parseChannel` is optional, and you need to add `Illuminate\Contracts\Foundaiton\Application` using DI for this to work. I would guess this should be the expected ways of doing this but feedbacks are welcome.